bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -524,24 +554,67 @@ impl Time {
     /// ```
     #[inline]
     pub fn to_consensus_u32(self) -> u32 { self.0 }
-}
 
-impl_parse_str_from_int_fallible!(Time, u32, from_consensus, Error);
+    fn from_consensus_i64(n: i64) -> Result<Self, ParseError> {
+        let n = u32::try_from(n).map_err(|_| ParseError::Conversion(n))?;
+        Self::from_consensus(n).map_err(Into::into)
+    }
+}
 
 impl fmt::Display for Time {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
 }
 
 impl FromHexStr for Time {
-    type Error = Error;
+    type Error = ParseTimeError;
 
     #[inline]
     fn from_hex_str_no_prefix<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, Self::Error> {
-        let time = crate::parse::hex_u32(s)?;
-        Time::from_consensus(time)
+        let height = i64::from_str_radix(s.as_ref(), 16)
+            .map_err(ParseError::invalid_int(s))?;
```

Comment:
I don't understand this change, why call `from_str_radix` on a signed type? I'm guessing you are trying to improve the error reporting coming from `std` but I don't see how?

Reply:
I see `// we use i64 to have nicer messages for negative values` but I don't see where we get negative values from?

Reply:
From parsing, it's in the PR description. For `-42` `std` says "invalid digit" which is confusing, I want to just say that the value is below the limit which is 0. And for that it's easiest to parse `i64` because if there actually is another invalid digit (`-42blocks`) then the error makes sense and `i64` losslessly holds `u32`. (Technically `i33` would be enough...)

Reply:
Should probably elaborate the comment here as well.

Reply:
There is a comment on the variant. I can copy it.

Reply:
> When minus sign is present std reports it as invalid digit which is less helpful than saying negative numbers are not allowed.

Ha! Guess I did not read the PR description close enough. A wee code comment on the `i64` would be great please.

Reply:
Oh you've pushed already - excuse the noise, I was mindlessly reading the thread.